### PR TITLE
Rewrote /user/logout to Go

### LINKED ISSUE #3841: Rewrite /user/logout to Go
## I'm submitting a ...
improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
- Documentation (possibly)
- Traffic Ops

## Current behavior:
`/user/logout` is handled by Perl

## Expected / new behavior:
`/user/logout` should be handled by Go.

### Minimum API Version
1.1

### Request Methods
- [ ] `POST`

## Minimal reproduction of the problem with instructions:
N/A

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR (Pull Request) do?

- [x] This PR fixes #3841 

Rewrote /user/logout to Go

## Which Traffic Control components are affected by this PR?
- Traffic Ops

functionality is not changing, so no docs

## What is the best way to verify this PR?
Authenticate as normal, then just logout. Normally, this would be done in Traffic Portal through the button on the sidebar. Then just observe that you're not authenticated - no endpoints (except `/ping` ) should respond with anything other than an "Unauthenticated" error

## The following criteria are ALL met by this PR
- [x] This PR includes tests
- [x] I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/login/logout_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+ * under the License.
+ */
+
+import "context"
+import "net/http"
+import "net/http/httptest"
+import "testing"
+import "time"
+
+// import "github.com/apache/trafficcontrol/lib/go-log"
```
Comment (mhoppa) [inline traffic_ops/traffic_ops_golang/login/logout_test.go]: nit remove ?

Also another nit should imports be in import block here and above?

File: traffic_ops/traffic_ops_golang/login/logout_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+		if parsedCookie.AuthData != testUser.UserName {
+			t.Errorf("Incorrect user parsed from cookie; expected '%s' but got: %s", testUser.UserName, parsedCookie.AuthData)
+		}
+	}
+
+	if !cookieFound {
+		t.Errorf("Expected handler to set the '%s' cookie, but it didn't", tocookie.Name)
+	}
+}
```
Comment (mhoppa) [inline traffic_ops/traffic_ops_golang/login/logout_test.go]: great test case!

File: traffic_ops/traffic_ops_golang/login/logout_test.go (hunk 1 above)
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/login/logout_test.go]: I don't like setting precedents for arbitrary stylistic changes without a style guide. As a Python programmer this style is most pleasing to me. I'm not saying I'd refuse to put them in blocks, just explaining my choice. If and when (hopefully when) there _is_ a style guide, I'll be more than happy to conform to it.
That comment should absolutely be removed, though.

Comment (mhoppa) [reply traffic_ops/traffic_ops_golang/login/logout_test.go]: 👍 completely agree on the style guide hence the comment being a nit/question and still approval of the PR.

FWIW per golangs _style guide_ they have the imports as a block https://github.com/golang/go/wiki/CodeReviewComments#imports

**Pull request merged**